File: frontend/src/lib/permissions.js
import { getRole } from './auth'

export function isSuperAdmin(role) {
  return (role || getRole()) === 'SUPER_ADMIN'
}

export function isDeptAdmin(role) {
  return (role || getRole()) === 'DEPARTMENT_ADMIN'
}

export function isManager(role) {
  return (role || getRole()) === 'MANAGER'
}

export function isUser(role) {
  return (role || getRole()) === 'USER'
}

export function isDeptAdminOrAbove(role) {
  const r = role || getRole()
  return r === 'SUPER_ADMIN' || r === 'DEPARTMENT_ADMIN'
}

export function isManagerOrAbove(role) {
  const r = role || getRole()
  return r === 'SUPER_ADMIN' || r === 'DEPARTMENT_ADMIN' || r === 'MANAGER'
}

File: assets/views/permissions.py
from rest_framework.permissions import BasePermission


class IsAuth(BasePermission):
    def has_permission(self, request, view):
        return request.user.is_authenticated


class IsManagerOrAbove(BasePermission):
    def has_permission(self, request, view):
        return request.user.is_authenticated and request.user.role in ("MANAGER", "DEPARTMENT_ADMIN", "SUPER_ADMIN")


class IsDeptAdminOrAbove(BasePermission):
    def has_permission(self, request, view):
        return request.user.is_authenticated and request.user.role in ("DEPARTMENT_ADMIN", "SUPER_ADMIN")


class IsSuperAdmin(BasePermission):
    def has_permission(self, request, view):
        return request.user.is_authenticated and request.user.role == "SUPER_ADMIN"
